podpac/core/node.py

```diff
@@ -748,3 +751,47 @@ def clear_disk_cache(self, attr='*', node_cache=False, all_cache=False):
         else:
             for f in glob.glob(self.cache_path(attr)):
                 os.remove(f)
+
+
+def node_eval(fn):
+    """
+    Decorator for Node eval methods that handles caching and a user provided output argument.
+
+    fn : function
+        Node eval method to wrap
+
+    Returns
+    -------
+    wrapper : function
+        Wrapped node eval method
+    """
+
+    cache_key = 'output'
+
+    @functools.wraps(fn)
+    def wrapper(self, coordinates, output=None):
+        if self.debug:
+            self._requested_coordinates = coordinates
+
+        cache_coordinates = coordinates.transform(sorted(coordinates.dims)) # order agnostic caching
```

Comment:
This should probably go into the caching framework directly? i.e. has_cache, put_cache, and get_cache, would work with any order, doing the transpose for you each time.